app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -214,9 +223,105 @@ class AgendaRepositoryImpl(
         return Resource.Success()
     }
 
-    override suspend fun deleteAllAgendaTables() {
+    override suspend fun syncFullAgenda() {
+        syncModifiedAgendaItems()
+        val results = getResourceResult { agendaApi.getFullAgenda() }
+        when (results) {
+            is Resource.Error -> {
+                Log.e(
+                    "Update Agenda of Date Error",
+                    results.message?.asString(appContext) ?: "Unknown Error"
+                )
+            }
+            is Resource.Success -> {
+                val localReminders = db.getReminderDao().getAllReminders().first()
+                localReminders.forEach { localReminder ->
+                    val containsLocalId = results.data?.reminders?.any {
+                        it.id == localReminder.id
+                    } == true
+                    if (!containsLocalId) {
+                        db.getReminderDao().deleteReminder(localReminder.id)
+                        cancelScheduledNotification(localReminder.id)
+                    }
+                }
+                results.data?.reminders?.forEach { reminderDto ->
+                    val localReminder = db.getReminderDao().getReminderById(reminderDto.id)
+                    val remoteReminder = reminderDto.toReminderEntity(
+                        isDone = localReminder?.isDone ?: false
+                    )
+                    db.getReminderDao().upsertReminder(remoteReminder)
+                    scheduleNotification(remoteReminder.toReminder())
+                }
```

Comment:
I'd just clear the whole DB before loading the new agenda items. Your approach might quickly lead to bugs if you try to detect which items changed etc